app/models/grade_entry_form.rb

```diff
@@ -93,25 +93,34 @@ def create_all_grade_entry_students
 
   def export_as_csv(role)
     if role.instructor?
-      students = Student.left_outer_joins(:grade_entry_students, :user)
+      students = Student.left_outer_joins(:grade_entry_students, :user, :section)
                         .where(hidden: false, 'grade_entry_students.assessment_id': self.id)
                         .order(:user_name)
-                        .pluck(:user_name, 'grade_entry_students.total_grade')
+                        .pluck_to_hash(:user_name, :last_name, :first_name, 'name as section_name',
+                                       :id_number, :email, 'grade_entry_students.total_grade')
     elsif role.ta?
+
       students = role.grade_entry_students
                      .joins(role: :user)
-                     .where(grade_entry_form: self, 'role.hidden': false, 'grade_entry_students.assessment_id': self.id)
+                     .joins('LEFT OUTER JOIN sections ON sections.id = roles.section_id')
+                     .where(grade_entry_form: self, 'roles.hidden': false,
+                            'grade_entry_students.assessment_id': self.id)
                      .order(:user_name)
-                     .pluck(:user_name, 'grade_entry_students.total_grade')
+                     .pluck_to_hash(:user_name, :last_name, :first_name, 'name as section_name',
+                                    :id_number, :email, 'grade_entry_students.total_grade')
+
     end
     headers = []
-    # The first row in the CSV file will contain the column names
-    titles = [''] + self.grade_entry_items.pluck(:name)
+    titles = Student::CSV_ORDER.map do |field|
```

Comment:
Use single-line block syntax (when I said "throughout" I meant across all files/changes, but perhaps you missed this file).